app/code/Magento/Customer/Model/Address.php

```diff
@@ -171,13 +171,14 @@ public function updateData(AddressInterface $address)
     public function getDataModel($defaultBillingAddressId = null, $defaultShippingAddressId = null)
     {
         if ($this->getCustomerId() || $this->getParentId()) {
-            if ($this->getCustomer()->getDefaultBillingAddress()) {
+            if ((intval($defaultBillingAddressId) == 0) && $this->getCustomer()->getDefaultBillingAddress()) {
```

Comment:
also there is possible bug in this logic , when  $defaultBillingAddressId is equal 0 , but not null. Your assertion still would be positive